### PR TITLE
Unfold plotting

### PR DESCRIPTION
**Pull request by rahulgrit**
Closed the previous pull request and creating a new one after rebasing. All changes are pertaining to plotting changes within RooFit.

@cburgard's comments from the previous MR 
The RooFit "plotOn" plotting engine is very useful to visualize RooFit objects.
However, there is currently a strict separation between "Data-like" objects (which can only really be drawn with points) and "Function-like" objects (which can only really be drawn as lines and areas).
However, in techniques like unfolding, "data" is often times corrected data, and hence might actually be a function-type object in RooFit.
With this PR, functionality is added to RooAbsReal::plotOn that allows to plot functions "data-like", with data points including error bars when using the draw option "P", which was previously unusable.

### CODE REVIEW

File: roofit/roofitcore/src/RooHist.cxx [hunk 1]
```diff
@@ -260,6 +261,152 @@ RooHist::RooHist(const RooHist& hist1, const RooHist& hist2, Double_t wgt1, Doub
...
+  funcPtr= f.bindVars(x,normVars,kTRUE);
+
+  // apply a scale factor if necessary
+  if(scaleFactor != 1) {
+    rawPtr= funcPtr;
+    funcPtr= new RooScaledFunc(*rawPtr,scaleFactor);
+  }
+  
+  // apply a scale factor if necessary
+  assert(0 != funcPtr);
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooHist.cxx]:
```suggestion
  assert(funcPtr);
```
More concise.

File: roofit/roofitcore/src/RooHist.cxx [hunk 2]
```diff
@@ -260,6 +261,152 @@ RooHist::RooHist(const RooHist& hist1, const RooHist& hist2, Double_t wgt1, Doub
...
+      title.Append("/ ");
+      title.Append(x.getUnit());
+      title.Append(" ");
+    }
+    title.Append(")");
+  }
+  setYAxisLabel(title.Data());
+
+  RooAbsFunc *funcPtr = 0;
+  RooAbsFunc *rawPtr  = 0;
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooHist.cxx]:
```suggestion
  RooAbsFunc *rawPtr  = nullptr;
```

File: roofit/roofitcore/src/RooHist.cxx [hunk 3]
```diff
@@ -260,6 +261,152 @@ RooHist::RooHist(const RooHist& hist1, const RooHist& hist2, Double_t wgt1, Doub
...
+    if(0 != strlen(x.getUnit())) {
+      title.Append("/ ");
+      title.Append(x.getUnit());
+      title.Append(" ");
+    }
+    title.Append(")");
+  }
+  setYAxisLabel(title.Data());
+
+  RooAbsFunc *funcPtr = 0;
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooHist.cxx]:
```suggestion
  RooAbsFunc *funcPtr = nullptr;
```
clang-format modernize will trigger here.

File: roofit/roofitcore/src/RooHist.cxx [hunk 4]
```diff
@@ -260,6 +261,152 @@ RooHist::RooHist(const RooHist& hist1, const RooHist& hist2, Double_t wgt1, Doub
...
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooHist.cxx]:
```suggestion
  RooAbsFunc *funcPtr = nullptr;
```

File: roofit/roofitcore/src/RooHist.cxx [hunk 5]
```diff
@@ -260,6 +261,152 @@ RooHist::RooHist(const RooHist& hist1, const RooHist& hist2, Double_t wgt1, Doub
...
+  RooAbsFunc *rawPtr  = 0;
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooHist.cxx]:
```suggestion
  RooAbsFunc *rawPtr  = nullptr;
```

File: roofit/roofitcore/src/RooHist.cxx [hunk 6]
```diff
@@ -260,6 +261,152 @@ RooHist::RooHist(const RooHist& hist1, const RooHist& hist2, Double_t wgt1, Doub
...
+  funcPtr= f.bindVars(x,normVars,kTRUE);
+
+  // apply a scale factor if necessary
+  if(scaleFactor != 1) {
+    rawPtr= funcPtr;
+    funcPtr= new RooScaledFunc(*rawPtr,scaleFactor);
+  }
+  
+  // apply a scale factor if necessary
+  assert(0 != funcPtr);
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooHist.cxx]:
```suggestion
  assert(funcPtr);
```

File: roofit/roofitcore/src/RooAbsReal.cxx [hunk 7]
```diff
@@ -1660,14 +1663,20 @@ void RooAbsReal::plotOnCompSelect(RooArgSet* selNodes) const
 /// <tr><td> `Components(const RooArgSet& compSet)` <td> As above, but pass a RooArgSet of the components themselves.
 ///
 /// <tr><th><th> Plotting control
-/// <tr><td> `DrawOption(const char* opt)`     <td> Select ROOT draw option for resulting TGraph object
+/// <tr><td> `DrawOption(const char* opt)`     <td> Select ROOT draw option for resulting TGraph object. Currently supported options are "F" (fill), "L" (line), and "P" (points). <i>Caution: Choosing option "P" will cause RooFit to plot (and treat) this pdf as if it were data! This is intended for plotting "corrected data"-type pdfs such as "data-minus-background" or unfolded datasets.</i>
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooAbsReal.cxx]: Can you break the line in two / three?
Try doxygen's
```
\note Choosing option "P"
or
\warning Choosing option "P"
```
Doxygen indents, colors, and adds either *Note* or *Warning* to the paragraph.

File: roofit/roofitcore/src/RooAbsReal.cxx [hunk 8]
```diff
@@ -1643,8 +1644,10 @@ void RooAbsReal::plotOnCompSelect(RooArgSet* selNodes) const
 /// <tr><td> `EvalErrorValue(Double_t value)`  <td> Set curve points at which (pdf) evaluation error occur to specified value. By default the
 ///                                    function value is plotted.
 ///
-/// <tr><td> `Normalization(Double_t scale, ScaleType code)`   <td> Adjust normalization by given scale factor. Interpretation of number depends on code: Relative:
-///                     relative adjustment factor, NumEvent: scale to match given number of events.
+/// <tr><td> `Normalization(Double_t scale, ScaleType code)`   <td> Adjust normalization by given scale factor. Interpretation of number depends on code:
+///                     Relative: relative adjustment factor for a normalized function,
+///                     NumEvent: scale to match given number of events.
+///                     Raw: relative adjustment factor for an un-normalized function.
```
Comment (hageboeck) [inline roofit/roofitcore/src/RooAbsReal.cxx]:
```
///                     - `Relative` Relative adjustment factor for a normalized function.
///                     - `NumEvent` Scale to match given number of events.
///                     - `Raw` Relative adjustment factor for an un-normalized function.
```
Like this, doxygen should typeset a list.

**Comment by cburgard** [comment]
all suggestions implemented now, @hageboeck please review!

**Comment by linev** [comment]
Please, do not use merge commits in PR.
Always rebase your branch before creating a PR

**Comment by rahulgrit** [comment]
Closing this PR to create a new one with the rebased code.

**Pull request closed without merging**